waspc/data/Generator/templates/server/src/routes/auth/providers/index.ts

```diff
@@ -0,0 +1,28 @@
+import { Router } from "express";
+
+import { getDirFromFileUrl, importJsFilesFromDir } from "../../../utils.js";
+
+import { ProviderConfig } from "./types";
+
+const providers = await importProviders();
+
+const router = Router();
+
+for (const provider of providers) {
+  const { init, setupRouter } = provider;
+  let initData;
+  if (init) {
+    initData = await init(provider);
+  }
+  const providerRouter = setupRouter(provider, initData);
+  router.use(`/${provider.slug}`, providerRouter);
+  console.log(`🚀 Init ${provider.name} auth provider`);
+}
+
+export default router;
+
+async function importProviders(): Promise<ProviderConfig[]> {
+  const currentExecutionDir = getDirFromFileUrl(import.meta.url);
+  const providers = await importJsFilesFromDir(currentExecutionDir, "./config");
```

Comment:
Since we came up with it during pairing, I selfishly vote for it. :D I think if we can avoid templates it will make our code much easier to understand and reason about. The conditional presence of a file makes sense to me for dictating on/off behavior.